1.0/docs/devguide/templates.md

```diff
@@ -277,6 +283,23 @@ different name for the index property.
     </template>
     {% endraw %}
 
+### Forcing synchronous renders {#synchronous-renders}
+
+Call [`render`](/{% polymer_version_dir %}/api/#dom-repeat:method-render) 
+to force a `dom-repeat` template to synchronously render any changes to its 
+data. Normally changes are batched and rendered asynchronously. This is 
+useful in two scenarios:
+
+*   For unit testing, to ensure items have rendered before checking the 
+    generated DOM.
+*   To ensure a list of items have rendered before scrolling to a 
+    specific item.
```

Comment:
*   To rerun the `sort` or `filter` functions when an a piece of data changes _outside_ the array (for example, sort order or filter criteria).